src/t8_forest/t8_forest_vtk.cxx

```diff
@@ -160,8 +165,16 @@ t8_forest_write_vtk_via_API (t8_forest_t forest, const char *fileprefix)
   /* 
    * The cellTypes Array stores the element types as integers(see vtk doc).
    */
-  int                *cellTypes =
-    T8_ALLOC (int, t8_forest_get_local_num_elements (forest));
+  num_elements = t8_forest_get_local_num_elements (forest);
+  int                *cellTypes = T8_ALLOC (int, num_elements);
+  /*
+   * We have to define the vtk UnvtkUnsignedCharArrays that hold 
+   * metadata if wanted. 
+   */
+  vtkUnsignedCharArray *vtk_treeid = vtkUnsignedCharArray::New ();
+  vtkUnsignedCharArray *vtk_mpirank = vtkUnsignedCharArray::New ();
+  vtkUnsignedCharArray *vtk_level = vtkUnsignedCharArray::New ();
+  vtkUnsignedCharArray *vtk_element_id = vtkUnsignedCharArray::New ();
```

Comment:
Do we have to use unsignedChar?
Or are there other options like Integer?

If it really uses `unsigned char` then
a) Our data types are signed and could (in theory) be negative. This would then cause weird underflow conversion results.
b) An unsigned char can only hold 255 different values and we certainly want to store larger numbers.

Reply:
Yes, i just confirmed it with VTK v9.0
It only stores values up to 255.

Reply:
I can change it to vtkTypeInt64Array or others. I'll do vtkTypeInt64Array for now, others are: vtkIntArray:The C++ standard does not define the exact size of the int type, so use of this type directly is discouraged. If an array of 32 bit integers is needed, prefer vtkTypeInt32Array to this class.

Reply:
In t8_vtk.h we define datatypes for the manual writing and distinguish between locidx and gloidx.
We should do something similar and add defines like
```
#if T8_WITH_VTK
#define t8_vtk_locidx_array_type vtkTypeInt32Array
#define t8_vtk_gloidx_array_type vtkTypeInt64Array
#endif
```
and then we can use t8_vtk_locidx_array_type and t8_vtk_gloidx_array_type.
We may also add assertions to ensure no type conversion errors occur...